### ISSUE TITLE
Truncate matching line for compressed/obfuscated single line files

### ISSUE THREAD

**Issue by letronje** [ggreer/the_silver_searcher#514]
For a web application, its normal to contain compressed js/css files that just have a single long line. 
For long lines, Is it possible to ask SS to only show the match and X chars before and after the match ?

**Comment by KyleAMathews** [comment]
:+1:

**Comment by cabello** [comment]
I could use that feature as well, when ag finds a match the entire file (because it's a one-liner) is printed.

**Comment by spiderpug** [comment]
I would love this as well. Using ag with vim, those minimized js/css files are annoying. If someone has a solution I would be interested!

**Comment by tweekmonster** [comment]
Yes, please add a flag to truncate the lines.  As a hopefully short term fix in Vim, I wrote a python script that ag can be piped into: https://gist.github.com/tweekmonster/86f043a07d68eaa062de

**Comment by chocmake** [comment]
Definitely would be appreciated. Experienced this today (also happened to be my first test with ag :p) and was a tad overwhelming with minified files (was searching a directory of Lua). Thought I'd capture a GIF showing the first three files and a handful of matches (kept going so I had to trim it).

![Example-optimized](https://user-images.githubusercontent.com/34178938/210123379-7ffabf29-b20d-47f3-8875-0a54f4bcde16.gif)
